### PR TITLE
Add debug output to kubectl-log-e2e.

### LINKED ISSUE #20287: kubectl log --since e2e test is hard to debug
I noticed this flake by looking at the submit queue.  The test is simple, but when it fails, we have no good way to debug it.

```
• Failure [12.004 seconds]
Kubectl client
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:992
  Kubectl logs
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:781
    should be able to retrieve and filter logs [Conformance] [It]
    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:780

    Expected
        <int>: 23
    to be <
        <int>: 23

    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:778
```

Below is the test in question.  When the test fails, we should print the lines we got in each case.

```go
        By("restricting to a time range")
        time.Sleep(1500 * time.Millisecond) // ensure that startup logs on the node are seen as older than 1s
        out = runKubectlOrDie("log", pod.Name, containerName, nsFlag, "--since=1s")
        recent := len(strings.Split(out, "\n"))
        out = runKubectlOrDie("log", pod.Name, containerName, nsFlag, "--since=24h")
        older := len(strings.Split(out, "\n"))
        Expect(recent).To(BeNumerically("<", older))
```

### PR DESCRIPTION
**Pull request by mml**
Fixes #20287.

### CODE REVIEW

File: test/e2e/kubectl.go [hunk 1]
```diff
@@ -771,10 +771,13 @@ var _ = Describe("Kubectl client", func() {
...
-				out = runKubectlOrDie("log", pod.Name, containerName, nsFlag, "--since=1s")
-				recent := len(strings.Split(out, "\n"))
-				out = runKubectlOrDie("log", pod.Name, containerName, nsFlag, "--since=24h")
-				older := len(strings.Split(out, "\n"))
+				recent_out := runKubectlOrDie("log", pod.Name, containerName, nsFlag, "--since=1s")
+				recent := len(strings.Split(recent_out, "\n"))
+				older_out := runKubectlOrDie("log", pod.Name, containerName, nsFlag, "--since=24h")
+				older := len(strings.Split(older_out, "\n"))
+				if recent >= older {
+					Failf("expected recent(%v) to be less than older(%v)\nrecent lines:\n%v\nolder lines:\n%v\n", recent, older, recent_out, older_out)
```
Comment (ixdy) [inline test/e2e/kubectl.go]: Just fold this into the `Expect()` line below - see https://onsi.github.io/gomega/#annotating-assertions for details.

**Comment by mml** [comment]
> Just fold this into the Expect() line below - see https://onsi.github.io/gomega/#annotating-assertions for details.

Done.

**Comment by ixdy** [comment]
LGTM, thanks!

**Comment by fabioy** [comment]
Manually merging.

**Pull request merged**